File: Problems/Algorithms/1512. Number of Good Pairs/good_pairs.py
#!/usr/bin/env python3

from collections import Counter
from typing import List


class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        counts = Counter(nums)
        ans = 0
        for value in counts.values():
            ans += value * (value - 1) // 2
        return ans

File: Problems/Algorithms/1512. Number of Good Pairs/good-pairs.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int ans = 0;
        unordered_map<int, int> count;
        for (int num : nums) {
            ans += count[num]++;
        }

        return ans;
    }
};
